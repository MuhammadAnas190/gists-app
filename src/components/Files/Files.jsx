import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { IconText } from '../IconText/IconText';
import { isEmpty } from '../../utils/helpers';

export const Files = ({ files }) => {
  /* checking if the `files` array is empty. If it is empty, it will return text "No files found" */
  if (isEmpty(files)) {
    return <p>No files found</p>;
  }

  return (
    <FilesWrapper>
      {files.map(({ filename, raw_url: rawUrl }) => (
        <IconText iconName="file" link={rawUrl} text={filename} key={filename} />
      ))}
    </FilesWrapper>
  );
};

const FilesWrapper = styled.div`
  display: inline-flex;
  gap: 0.5rem;
  align-items: center;
  font-weight: 500;
  flex-wrap: wrap;
`;

Files.propTypes = {
  files: PropTypes.arrayOf(
    PropTypes.shape({
      filename: PropTypes.string.isRequired,
      raw_url: PropTypes.string.isRequired,
    })
  ).isRequired,
};
