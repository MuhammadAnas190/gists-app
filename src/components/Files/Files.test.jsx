import { render } from '@testing-library/react';
import { Files } from './Files';

describe('Files Component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Files files={[{ filename: 'name', raw_url: 'https://google.com' }]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
