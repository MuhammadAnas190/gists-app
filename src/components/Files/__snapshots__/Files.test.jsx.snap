// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Files Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-eqUAAy itNODb"
  >
    <div
      class="sc-aXZVg fxkAHn"
    >
      <span
        class="octicon octicon-file"
      />
      <a
        class="sc-gEvEer dAbXZQ"
        href="https://google.com"
        target="_blank"
      >
        name
      </a>
    </div>
  </div>
</DocumentFragment>
`;
